protostar-cli/src/starknet_commands/declare.rs

```diff
@@ -0,0 +1,49 @@
+use anyhow::{Error, Result};
+use camino::Utf8PathBuf;
+use clap::Args;
+use starknet::{
+    accounts::{Account, SingleOwnerAccount},
+    core::types::contract::{CompiledClass, SierraClass},
+    providers::jsonrpc::{HttpTransport, JsonRpcClient},
+    signers::LocalWallet,
+};
+use std::sync::Arc;
+use starknet::core::types::DeclareTransactionResult;
+
+#[derive(Args)]
+#[command(about = "Declare a contract to starknet", long_about = None)]
+pub struct Declare {
+    /// Path to the sierra compiled contract
+    #[clap(short = 's', long = "sierra-contract-path")]
+    pub sierra_contract_path: Utf8PathBuf,
+
+    /// Path to the casm compiled contract
+    #[clap(short = 'c', long = "casm-contract-path")]
+    pub casm_contract_path: Utf8PathBuf,
+}
+
+pub async fn declare(
+    sierra_contract_path: &Utf8PathBuf,
+    casm_contract_path: &Utf8PathBuf,
+    account: &mut SingleOwnerAccount<&JsonRpcClient<HttpTransport>, LocalWallet>,
+) -> Result<DeclareTransactionResult, Error> {
+    let contract_definition: SierraClass = {
+        let file_contents = std::fs::read(sierra_contract_path)
+            .map_err(|err| anyhow::anyhow!("Failed to read contract file: {}", err))?;
```

Comment:
```suggestion
        let file_contents = std::fs::read(sierra_contract_path)
            .with_context(|| format!("Failed to read contract file: {}", err))?;
```

Reply:
Nobody asked, but about `context` vs `with_context`. `with_context` is lazily evaluated so it runs only on error, so you won't be doing costly string formatting if not necessary.